src/lib/operators/join_hash.cpp

```diff
@@ -416,33 +418,48 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
           auto pos_list_left_local = std::make_shared<PosList>();
           auto pos_list_right_local = std::make_shared<PosList>();
 
-          for (size_t i = partition_begin; i < partition_end; ++i) {
-            auto& row = partition[i];
+          if (_mode == JoinMode::Anti) {
+            // find all rows that do NOT match
+            for (size_t i = partition_begin; i < partition_end; ++i) {
+              auto& row = partition[i];
+              auto row_ids = hashtable->get(row.value);
 
-            if (_mode == JoinMode::Inner && row.row_id.chunk_offset == INVALID_CHUNK_OFFSET) {
-              continue;
+              if (!row_ids) {
+                pos_list_right_local->emplace_back(row.row_id);
+              }
             }
 
-            auto row_ids = hashtable->get(row.value);
+            pos_list_left[current_partition_id] = pos_list_left_local;
+            pos_list_right[current_partition_id] = pos_list_right_local;
+          } else {
```

Comment:
Feels like we could add `return` above and decrease the indentation level. Or maybe even split the different join types into their own methods.